component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/introspection/IntrospectionResponseInterceptor.java

```diff
@@ -74,15 +75,17 @@ Omit the cert thumbprint scope (added by MTLSTokenBindingAuthorizationCodeGrantH
             oAuth2IntrospectionResponseDTO.setScope(String.join(" ", scopeList));
         }
 
-        Map<String, Object> introspectionResponseProperties = oAuth2IntrospectionResponseDTO.getProperties();
-        if (introspectionResponseProperties == null) {
-            introspectionResponseProperties = new HashMap<>();
-        }
+        if (Boolean.parseBoolean(IdentityUtil.getProperty(CommonConstants.ENABLE_TLS_CERT_TOKEN_BINDING))) {
```

Comment:
This should be done only if the config is false right? also lets update the constant name to match the config

Reply:
Lets add a comment mentioning the reason(new feature, preserving backward compatibility, etc.) to check this config